### PR TITLE
Adding all of Alexia's work to the app (expression sheet parsing)

### PR DESCRIPTION
**Pull request by afiller1**
See entire series of issue #734 . Added a few checks for expression sheet parsing as well as unit tests. Refactored the way that additional sheets (aside from network sheets) are parsed. Parsing all happens in spreadsheet-controller.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 GRNsight
 ========
 [![DOI](https://zenodo.org/badge/16195791.svg)](https://zenodo.org/badge/latestdoi/16195791)
-[![Build Status](https://travis-ci.org/dondi/GRNsight.svg?branch=beta)](https://travis-ci.org/dondi/GRNsight)
-[![Coverage Status](https://coveralls.io/repos/github/dondi/GRNsight/badge.svg?branch=beta)](https://coveralls.io/github/dondi/GRNsight?branch=beta)
+[![Build Status](https://travis-ci.org/dondi/GRNsight.svg?branch=master)](https://travis-ci.org/dondi/GRNsight)
+[![Coverage Status](https://coveralls.io/repos/github/dondi/GRNsight/badge.svg?branch=master)](https://coveralls.io/github/dondi/GRNsight?branch=beta)
```
Comment (dondi) [inline README.md]: Should stay as beta.

File: server/controllers/additional-sheet-parser.js [hunk 2]
```diff
@@ -8,20 +8,20 @@ var TWO_COL_SHEET_NAMES = [
...
-var isExpressionSheet = function (sheetName) {
-    return EXPRESSION_SHEET_SUFFIXES.some(function (suffix) {
-        return sheetName.includes(suffix);
-    });
-};
+// var isExpressionSheet = function (sheetName) {
+//     return EXPRESSION_SHEET_SUFFIXES.some(function (suffix) {
+//         return sheetName.includes(suffix);
+//     });
+// };
```
Comment (dondi) [inline server/controllers/additional-sheet-parser.js]: Please eliminate commented-out code.

File: server/controllers/exporters/xlsx.js [hunk 3]
```diff
@@ -83,6 +80,7 @@ const buildXlsxSheet = function (network) {
         }
     );
 
+// This is the code that causes crash in Excel export.
```
Comment (dondi) [inline server/controllers/exporters/xlsx.js]: Is this comment still true?

File: server/controllers/spreadsheet-controller.js [hunk 4]
```diff
@@ -440,8 +461,52 @@ var processGRNmap = function (path, res, app) {
...
+    var expressionData = parseExpressionSheets(sheet);
+
+    // Add errors and warnings from meta sheets
+    if(additionalData['meta']['errors'] !== undefined) {
+        for(var i = 0; i < additionalData['meta']['errors'].length; i++) {
+            network['errors'].push(additionalData['meta']['errors'][i]);
+        }
+        for(var i = 0; i < additionalData['meta']['warnings'].length; i++) {
+            network['errors'].push(additionalData['meta']['warnings'][i]);
+        }
```
Comment (dondi) [inline server/controllers/spreadsheet-controller.js]: Modernization note: consider using an iterator function like `forEach`.

File: test/test.js [hunk 5]
```diff
@@ -188,6 +188,58 @@ var emptyRowError = function (input, frequency) {
...
+var idLabelError = function (input, frequency) {
+    var sheet = xlsx.parse(input);
+    var network = parseExpressionSheet(sheet);
+    assert.equal(frequency, network.expression.wt_log2_expression.errors.length);
+
+    for (var i = 0; i < frequency; i++) {
+        assert.equal(
+      "MISLABELED_ID_CELL",
+      network.expression.wt_log2_expression.errors[i].errorCode
+    );
```
Comment (dondi) [inline test/test.js]: Double-check/fix indentation.

File: web-client/public/js/setup-load-and-import-handlers.js [hunk 6]
```diff
@@ -70,7 +70,9 @@ const networkErrorDisplayer = xhr => {
 
     if (!err.errors) { // will be falsy if an error was thrown before the network was generated
         errorString += err;
+        console.log("First if");
     } else {
+        console.log("Second if");
```
Comment (dondi) [inline web-client/public/js/setup-load-and-import-handlers.js]: These console logs can go away.

File: web-client/public/js/warnings.js [hunk 7]
```diff
@@ -12,7 +11,7 @@ export var displayWarnings = function (warnings) {
 
     // Fill printed with 0s programatically
     var printed = [];
-    for (var i = 0; i < NUM_POSSIBLE_WARNINGS; i++) {
+    for (var i = 0; i <= NUM_POSSIBLE_WARNINGS; i++) {
```
Comment (dondi) [inline web-client/public/js/warnings.js]: Double check this?

File: README.md (hunk 1 above)
Comment (afiller1) [reply README.md]: Resolved by Mihir.

**Pull request merged**